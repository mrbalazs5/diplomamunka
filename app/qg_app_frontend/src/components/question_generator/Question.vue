<script setup lang="ts">
  import IconCopyToClipboard from '../icons/IconCopyToClipboard.vue'
  import Tooltip from "@/components/tooltip/Tooltip.vue";
</script>

<template>
  <div class="question">
    <span class="content">
      {{ text }}
    </span>
    <Tooltip text="Copy to clipboard">
      <IconCopyToClipboard class="copy-btn" @click="copyToClipboard()" />
    </Tooltip>
  </div>
</template>

<script lang="ts">
  import useClipboard from "@/utils/useClipboard";

  const { toClipboard } = useClipboard();

  export default {
    name: "Question",
    props: {
      text: String
    },
    methods: {
      copyToClipboard: async function (loginData) {
        try {
          await toClipboard(this.text);
        } catch (e) {
          console.error("Failed to copy question", e);
        }
      }
    }
  }
</script>

<style lang="scss" scoped>
  .question {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom: 10px;
    font-size: 18px;
  }

  .copy-btn {
    display: inline-block;
    margin-left: 10px;
    width: 20px;
    transition: 0.5s;
    stroke: #088522;
    cursor: pointer;

    &:hover {
      stroke: #ffffff;
    }
  }
</style>