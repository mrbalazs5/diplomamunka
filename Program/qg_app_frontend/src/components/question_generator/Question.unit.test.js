import { shallowMount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import Question from "@/components/question_generator/Question.vue";
test('mount component', () => {
    const wrapper = shallowMount(Question, {
        props: {
            value: 'test',
        },
        global: {
            stubs: ['a-select', 'a-select-option'],
        },
    });
    expect(wrapper.html()).toMatchSnapshot();
});
