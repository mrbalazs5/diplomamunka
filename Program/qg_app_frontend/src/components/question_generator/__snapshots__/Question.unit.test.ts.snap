// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `
"<div class=\\"question\\" data-v-38b2495b=\\"\\"><span class=\\"content\\" data-v-38b2495b=\\"\\">test</span>
  <tooltip-stub text=\\"Copy to clipboard\\" data-v-38b2495b=\\"\\"></tooltip-stub>
</div>"
`;
